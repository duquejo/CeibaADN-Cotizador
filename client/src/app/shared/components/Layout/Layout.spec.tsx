import * as React from 'react';
import { ShallowWrapper, mount, render, shallow } from 'enzyme';
import { Layout } from './index';

jest.mock('GestionCalendarios', () => {
    return ( <span>GestionCalendarios</span> );
}, { virtual: true });

describe('Pruebas sobre componente <Layout />', () => {

    let componentWrapper: ShallowWrapper;
    let componentProps: React.ComponentProps<typeof Layout>;

    beforeEach( () => {
        componentProps = {
            title: 'Página de prueba',
            description: 'Esta es la descripción de una página de prueba',
            children: require( 'GestionCalendarios' )
        };
        componentWrapper = shallow(<Layout { ...componentProps } />);
    });

    it('Debería renderizar el elemento', () => {
        expect( componentWrapper ).toMatchSnapshot();
    });
});