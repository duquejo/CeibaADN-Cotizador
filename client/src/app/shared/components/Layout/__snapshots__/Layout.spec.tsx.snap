// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre componente <Layout /> Debería renderizar el elemento 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Cotizador de Centros Vacacionales | 
      Página de prueba
    </title>
    <meta
      content="Esta es la descripción de una página de prueba"
      name="description"
    />
  </HelmetWrapper>
  <styled.div>
    <span>
      GestionCalendarios
    </span>
  </styled.div>
  <FooterContainer />
</Fragment>
`;
